File: cpp/threads/threads.cpp
#include <iostream>
#include <thread>
#include <vector>

void increment() {
    double counter = 0;
    for (int i = 0; i < 1000000000; i++) {
        counter++;
    }

    std::cout << "Counter is: " << counter << "\n";
}

int main() {
    std::vector<std::thread> threads;
    for (int i = 0; i < 1000; i++) {
        threads.push_back(std::thread(increment));
    }

    for (auto& t : threads) {
        t.join();
    }

    return 0;
}

File: rust/threads/threads.rs
use std::thread;

fn increment() {
    let mut counter:f64 = 0.0;
    let mut i:i32 = 0;
    while i < 1000000000 {
        counter += 1.0;
        i += 1;
    }

    println!("Counter is: {}", counter);
}

fn main() {
    let mut handles = vec![];

    let mut i:i32 = 0;
    while i < 1000 {
        let handle = thread::spawn( || {
            increment();
        });
        handles.push(handle);
        i += 1;
    }

    for handle in handles {
        handle.join().unwrap();
    }
}